src/radical/pilot/agent/rm/spark.py

```diff
@@ -0,0 +1,57 @@
+
+__copyright__ = "Copyright 2016, http://radical.rutgers.edu"
+__license__   = "MIT"
+
+
+import os
+import subprocess
+import multiprocessing
+
+from base import LRMS
+
+
+#===============================================================================
+
+class Spark(LRMS):
+
+    # --------------------------------------------------------------------------
+    #
+    def __init__(self, cfg, logger):
+
+        LRMS.__init__(self, cfg, logger)
+
+
+    # --------------------------------------------------------------------------
+    #
+    def _configure(self):
+
+        self._log.info("Using SPARK on localhost.")
+
+        selected_cpus = self.requested_cores
+
+        # when we profile the agent, we fake any number of cores, so don't
+        # perform any sanity checks.  Otherwise we use at most all available
+        # cores (and informa about unused ones)
+        if 'RADICAL_PILOT_PROFILE' not in os.environ:
+
+            detected_cpus = multiprocessing.cpu_count()
+
+            if detected_cpus < selected_cpus:
+                self._log.warn("insufficient cores: using available %d instead of requested %d.",
+                        detected_cpus, selected_cpus)
+                selected_cpus = detected_cpus
+
+            elif detected_cpus > selected_cpus:
+                self._log.warn("more cores available: using requested %d instead of available %d.",
+                        selected_cpus, detected_cpus)
+
+
+   
+
```

Comment:
Can we remove some of the empty lines, please?

(I would never have thought I'd ever make this comment on code :P )